Packet++/src/VrrpLayer.cpp

```diff
@@ -0,0 +1,533 @@
+		toStr += std::to_string(getAuthType());
+		toStr += ")";
+
+		return toStr;
+	}
+
+	uint8_t VrrpV2Layer::getAdvInt() const {
+		auto *packet = (vrrp_packet *) getData();
+		uint16_t authAdvInt = packet->authTypeAdvInt;
+		auto *authAdvIntPtr = (vrrpv2_auth_adv *) (&authAdvInt);
+		return authAdvIntPtr->advInt;
+	}
+
+	uint8_t VrrpV2Layer::getAuthType() const {
+		auto *packet = (vrrp_packet *) getData();
+		uint16_t authAdvInt = packet->authTypeAdvInt;
+		auto *authAdvIntPtr = (vrrpv2_auth_adv *) (&authAdvInt);
+		return authAdvIntPtr->authType;
+	}
+
+	uint16_t VrrpV2Layer::fillChecksum() {
+		auto *vrrpPacket = (vrrp_packet *) getData();
+		ScalarBuffer<uint16_t> buffer = {};
+		buffer.buffer = (uint16_t *) getData();
+		buffer.len = getHeaderLen();
+
+		vrrpPacket->checksum = 0;
+		uint16_t checksum = computeChecksum(&buffer, 1);
+		vrrpPacket->checksum = htobe16(checksum);
+
+		return checksum;
+	}
+
+	uint16_t VrrpV2Layer::calculateChecksum() const {
+		if ((getData() == nullptr) || (getDataLen() == 0)) {
+			return 0;
+		}
+
+		auto *vrrpPacketCopy = (vrrp_packet *) new uint8_t[getDataLen()];
+		memcpy(vrrpPacketCopy, getData(), getDataLen());
+
+		ScalarBuffer<uint16_t> buffer = {};
+		buffer.buffer = (uint16_t *) vrrpPacketCopy;
+		buffer.len = getHeaderLen();
+
+		vrrpPacketCopy->checksum = 0;
+		uint16_t checksum = computeChecksum(&buffer, 1);
+		delete[] vrrpPacketCopy;
+
+		return checksum;
+	}
+
+	/*************
+	 * Vrrpv3Layer
+	 *************/
+	std::string VrrpV3Layer::getAuthTypeDesc() const {
+		std::string toStr;
+
+		return toStr;
+	}
+
+	uint8_t VrrpV3Layer::getAdvInt() const {
+		auto *packet = (vrrp_packet *) getData();
+		uint16_t authAdvInt = packet->authTypeAdvInt;
+		auto *rsvdAdv = (vrrpv3_rsvd_adv *) (&authAdvInt);
+		return rsvdAdv->maxAdvInt;
+	}
+
+	uint16_t VrrpV3Layer::fillChecksum() {
+		auto *vrrpPacket = (vrrp_packet *) getData();
+		uint16_t checksum = calculateChecksum();
+
+		vrrpPacket->checksum = htobe16(checksum);
+
+		return checksum;
+	}
+
+	uint16_t VrrpV3Layer::calculateChecksum() const {
+		auto *ipLayer = m_Packet->getLayerOfType<pcpp::IPLayer>();
+		if (ipLayer == nullptr) {
+			PCPP_LOG_ERROR("Calculate checksum failed, for can not get IPLayer" << "");
+			return 0;
+		}
+
+		auto *vrrpPacketCopy = (vrrp_packet *) new uint8_t[getDataLen()];
+		memcpy(vrrpPacketCopy, getData(), getDataLen());
+		vrrpPacketCopy->checksum = 0;
+
+		pcpp::IPAddress srcIPAddr = ipLayer->getSrcIPAddress();
+		pcpp::IPAddress dstIPAddr = ipLayer->getDstIPAddress();
+		uint16_t checksum;
+		if (isIPv4Packet()) {
+			checksum = computePseudoHdrChecksum((uint8_t *) vrrpPacketCopy, getDataLen(), IPAddress::IPv4AddressType,
+												PACKETPP_IPPROTO_VRRP, srcIPAddr, dstIPAddr);
+		} else {
+			checksum = computePseudoHdrChecksum((uint8_t *) vrrpPacketCopy, getDataLen(), IPAddress::IPv6AddressType,
+												PACKETPP_IPPROTO_VRRP, srcIPAddr, dstIPAddr);
+		}
+
+		delete[]vrrpPacketCopy;
+
+		return checksum;
```

Comment:
Please see my previous comments in `VrrpV2Layer` regarding these 2 methods

Reply:
OK. I will fix it.